sdks/python/apache_beam/examples/snippets/transforms/element_wise/regex.py

```diff
@@ -0,0 +1,161 @@
+# coding=utf-8
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from __future__ import absolute_import
+from __future__ import print_function
+
+
+def regex_matches(test=None):
+  # [START regex_matches]
+  import apache_beam as beam
+  import re
+
+  def parse_plant(text):
+    m = re.match(r'^([^\s-]+)\s*-\s*(\w+)\s*-\s*(?P<duration>\w+)$', text)
+    if m:
+      yield {
+          'match': m.group(0),              # contains the entire matched text
+          'icon': m.group(1),               # ([^\s-]+) - group
+          'name': m.group(2),               # (\w+) - group
+          'duration': m.group('duration'),  # (?P<duration>\w+) - named group
+      }
+
+  with beam.Pipeline() as pipeline:
+    plant_matches = (
+        pipeline
+        | 'Garden plants' >> beam.Create([
+            '🍓   -   Strawberry   -   perennial',
+            '🥕 - Carrot - biennial',
+            '# 🍌 - invalid - format',
+            '🍆\t-\tEggplant\t-\tperennial',
+            '🍅 - Tomato - annual',
+            '🍉 - invalid - format with trailing words',
+            '🥔-Potato-perennial',
+        ])
+        | 'Parse plants' >> beam.FlatMap(parse_plant)
+        | beam.Map(print)
+    )
+    # [END regex_all_matches]
+    if test:
+      test(plant_matches)
+
+
+def regex_find(test=None):
```

Comment:
I would suggest regex_search to follow the same naming convention with re.search.

Reply:
Sure, I was following the Java version conventions, but it makes more sense to make this more Python-specific.